### DISCUSSION TITLE
How to implement shadow dom?

### DISCUSSION THREAD

**Discussion by go4cas** [unocss/unocss#1899] [Q&A]
I have the following in `vite.config.js`:
```javascript
export default {
  plugins: [
    Unocss({
      mode: 'shadow-dom',
      presets: [
        presetUno(),
        presetWebFonts({
          provider: 'google',
          fonts: {
            poppins: 'Poppins',
          },
        }),
      ],
    }),
  ],
}
```
Have removed `import 'uno.css'` from index.js.
This is an extract of the rendered markup:
<img width="1072" alt="Screenshot 2022-11-22 at 14 24 16" src="https://user-images.githubusercontent.com/12674784/203313678-0cbdfca0-63cb-4c6b-a3c6-007605c39924.png">

When the `mode: 'shadow-dom'` is added, and the `import 'uno.css'` removed, none of the elements (either inside or outside the shadow-dom) have the uno.css styles applied.

When the `import 'uno.css'` is added and the `mode: 'shadow-dom'` removed, the elements outside of the shadow-dom is rendered correctly.

Am I missing some config?

**Comment by NMTuan** [comment]
```vue
<script setup>
import { ref, onMounted } from 'vue'
const dom = ref('')
const style = document.createElement('style')
style.textContent = `@unocss-placeholder`
onMounted(() => {
    console.log(dom.value.appendChild(style))
})
</script>
```

https://github.com/NMTuan/base_vite_chrome_extension/blob/feature/20221122/src/content/Content.vue
